### PR TITLE
add popup code

### PR DESCRIPTION
**Pull request by NoSloppy**

### CODE REVIEW

File: style_editor.html [hunk 1]
```diff
@@ -8830,6 +8878,42 @@
 var use_graflex = true;
 var bad_fps = 0;
 
+</script>
+
+<div id="popup_window" class="popup-window">
+  <p id="popup_message"></p>
+  <button title="OK" onclick="dismissPopupMessage()">OK</button>
+  <br>
+  <label title="Disable this popup again. Refresh page to show again.">
```
Comment (profezzorn) [inline style_editor.html:8887]: "Disable this popup again"
What does that mean?

File: style_editor.html [hunk 2]
```diff
@@ -8830,6 +8878,42 @@
...
+      overlay.classList.add("show");
+    }
+  }
+  function dismissPopupMessage() {
+    popupWindow.classList.remove("show");
+    overlay.classList.remove("show");
+  }
+  function DontShowAgain() {
+    enablePopup = false;
+    // saveCookie("allowPopup", enablePopup);
```
Comment (profezzorn) [inline style_editor.html:8914]: Why commented out?
Also, if there is ever another type of poup, maybe  we need to save the "don't show again" separately for each type?

File: style_editor.html [hunk 3]
```diff
@@ -8839,6 +8923,8 @@
      if (bad_fps) bad_fps --;
   }
   if (bad_fps > 10 && use_graflex) {
+    popupMessage = "Struggling to render hilt model.<br>Switching to simpler design.<br>To re-enable Graflex model, go to Settings.";
```
Comment (profezzorn) [inline style_editor.html:8926]: Does this work? (I don't see where the popupMessage variable is created.)
Also, there doesn't seem to be a need for a variable, just put the string in the showPopupMessage() call, right?

File: style_editor.html (hunk 2 above)
Comment (NoSloppy) [reply style_editor.html:8914]: Commented as a placeholder for the saveCookie function that is yet to come. removed for now.
I'll work on adding a popupIdentifier or something.

Comment (profezzorn) [reply style_editor.html:8914]: Seems like we should just wait until we have saveCookie?

File: style_editor.html (hunk 1 above)
Comment (NoSloppy) [reply style_editor.html:8887]: bad edit. used to be "Don't show this popup again."
Then added the Refresh part, changed "Don't show" to "Disabled", but left the "again."

File: style_editor.html (hunk 3 above)
Comment (NoSloppy) [reply style_editor.html:8926]: oh yeah. that was silly.

File: style_editor.html (hunk 2 above)
Comment (NoSloppy) [reply style_editor.html:8914]: yes, there are other things to fix too. I'll resubmit after settings.

**Comment by NoSloppy** [comment]
no need to wait for another function

**Comment by profezzorn** [comment]
Conflicts....

**Comment by NoSloppy** [comment]
resolved.

**Pull request merged**